File: src/mffs/event/BlockInventoryDropDelayedEvent.java
package mffs.event;

import java.util.ArrayList;

import mffs.IDelayedEventHandler;
import mffs.base.TileEntityInventory;
import net.minecraft.block.Block;
import net.minecraft.item.ItemStack;
import net.minecraft.world.World;
import universalelectricity.api.vector.Vector3;

public class BlockInventoryDropDelayedEvent extends BlockDropDelayedEvent
{
	private TileEntityInventory projector;

	public BlockInventoryDropDelayedEvent(IDelayedEventHandler handler, int ticks, Block block, World world, Vector3 position, TileEntityInventory projector)
	{
		super(handler, ticks, block, world, position);
		this.projector = projector;
	}

	@Override
	protected void onEvent()
	{
		if (!this.world.isRemote)
		{
			if (this.position.getBlockID(this.world) == this.block.blockID)
			{
				ArrayList<ItemStack> itemStacks = this.block.getBlockDropped(this.world, this.position.intX(), this.position.intY(), this.position.intZ(), this.position.getBlockMetadata(world), 0);

				for (ItemStack itemStack : itemStacks)
				{
					this.projector.mergeIntoInventory(itemStack);
				}

				this.position.setBlock(this.world, 0);
			}
		}
	}
}

File: src/main/scala/mffs/mobilize/event/BlockInventoryDropDelayedEvent.scala
package mffs.mobilize.event

import mffs.base.TileMFFSInventory
import net.minecraft.block.Block
import net.minecraft.init.Blocks
import net.minecraft.world.World
import universalelectricity.core.transform.vector.Vector3

import scala.collection.JavaConversions._

class BlockInventoryDropDelayedEvent(handler: IDelayedEventHandler, ticks: Int, block: Block, world: World, position: Vector3, projector: TileMFFSInventory) extends BlockDropDelayedEvent(handler, ticks, block, world, position)
{
  protected override def onEvent
  {
    if (!world.isRemote)
    {
      if (position.getBlock(this.world) == block)
      {
        val itemStacks = block.getDrops(this.world, this.position.xi, this.position.yi, this.position.zi, this.position.getBlockMetadata(world), 0)

        for (itemStack <- itemStacks)
        {
          projector.mergeIntoInventory(itemStack)
        }

        position.setBlock(world, Blocks.air)
      }
    }
  }
}